core/src/main/java/org/apache/sling/testing/mock/sling/MockXSSAPIImpl.java

```diff
@@ -127,32 +128,32 @@ public String getValidXML(@Nullable String xml, @Nullable String defaultXml) {
 
     @Override
     public @Nullable String encodeForHTML(@Nullable String source) {
-        return source;
+        return source == null ? null : Encode.forHtml(source);
     }
 
     @Override
     public @Nullable String encodeForHTMLAttr(@Nullable String source) {
-        return source;
+        return source == null ? null : Encode.forHtmlAttribute(source);
     }
 
     @Override
     public @Nullable String encodeForXML(@Nullable String source) {
-        return source;
+        return source == null ? null : Encode.forXml(source);
     }
 
     @Override
     public @Nullable String encodeForXMLAttr(@Nullable String source) {
-        return source;
+        return source == null ? null : Encode.forXmlAttribute(source);
     }
 
     @Override
     public @Nullable String encodeForJSString(@Nullable String source) {
-        return source;
+        return source == null ? null : Encode.forJavaScript(source).replace("\\-", "\\u002D");
```

Comment:
Would be nice to not do a regex replace here (`String.replace` accepts a regex as parameter) + it replaces only the first hyphen. Then again, why is this hyphen replace needed?

Reply:
It was part of https://issues.apache.org/jira/browse/SLING-5946 and this is how it is done in https://github.com/apache/sling-org-apache-sling-xss/blame/b4c305a741cfe5a9ebad0e3c63f104ad1768d2e9/src/main/java/org/apache/sling/xss/impl/XSSAPIImpl.java#L401